Content.Server/Clothing/Systems/ChameleonClothingSystem.cs

```diff
@@ -94,6 +118,36 @@ private void UpdateUi(EntityUid uid, ChameleonClothingComponent? component = nul
         Dirty(uid, component);
     }
 
+    public string GetRandomValidPrototype(EntityUid uid, SlotFlags slot)
+    {
+        var variants = GetValidTargets(slot);
+        var clothPickIndex = _random.Next(variants.Count());
+        var pick = variants.ToList()[clothPickIndex];
+
+        return pick;
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        // Randomize EMP-affected clothing
+        var query = EntityQueryEnumerator<EmpDisabledComponent, ChameleonClothingComponent>();
+        while (query.MoveNext(out var uid, out var emp, out var chameleon))
+        {
+            if (_timing.CurTime < emp.DisabledUntil && chameleon.EmpContinious)
```

Comment:
is this check even necessary? Doesn't `EmpDisabledComponent` get removed once DisabledUntil is reached anyways? I don't think you need to check the time twice here.